metadeploy/api/jobs.py

```diff
@@ -296,3 +261,72 @@ def expire_preflights():
 
 
 expire_preflights_job = job(expire_preflights)
+
+
+class FakeUser:
+    def __init__(self, token):
+        self.token = token
+
+    def __bool__(self):  # pragma: nocover
+        return False
+
+
+def create_scratch_org(*, plan_id, email, org_name, result_id):
+    plan = Plan.objects.get(id=plan_id)
+    result = ScratchOrgJob.objects.get(id=result_id)
+    repo_url = plan.version.product.repo_url
+    repo_owner, repo_name = extract_user_and_repo(repo_url)
+    commit_ish = plan.commit_ish
+    try:
+        with local_github_checkout(
+            repo_owner, repo_name, commit_ish=commit_ish
+        ) as repo_root:
+            scratch_org_config, _, org_config = create_scratch_org_on_sf(
+                repo_owner=repo_owner,
+                repo_name=repo_name,
+                repo_url=repo_url,
+                repo_branch=commit_ish,
+                email=email,
+                project_path=repo_root,
+                org_name=org_name,
+            )
+    except Exception as e:
+        result.fail(e)
+        return
+
+    result.complete(scratch_org_config)
+    fake_user = FakeUser(token=(org_config.access_token, org_config.refresh_token))
+
+    # @@@ TODO: run a preflight if exists.
+    if plan.requires_preflight:
+        preflight_result = PreflightResult.objects.create(
+            user=None,
+            plan=plan,
+            organization_url=org_config.instance_url,
+            org_id=scratch_org_config["org_id"],
+        )
+        preflight(preflight_result.pk, forced_user=fake_user)
+
+    job = Job.objects.create(
+        user=None,
+        plan=plan,
+        organization_url=org_config.instance_url,
+        is_public=True,
+        org_id=scratch_org_config["org_id"],
+        uuid=result.uuid,
+    )
+
+    rq_job = run_flows.delay(
+        user=fake_user,
+        plan=plan,
+        skip_steps=[],
+        organization_url=org_config.instance_url,
+        result_class=Job,
+        result_id=job.id,
+    )
```

Comment:
@wlonk I don't think we want to actually be starting the installation automatically, at least not on the PR. Can we comment this out for now?